modules/functions.js

```diff
@@ -0,0 +1,92 @@
+export const taskEntry = document.getElementById('txtTask');
+export const tasks = [];
+const setStorage = () => {
+  localStorage.setItem('taskData', JSON.stringify(tasks));
+};
+const taskPopulate = (tasks) => {
+  const taskList = document.getElementById('container');
+  taskList.innerHTML = tasks.map((task) => `
+    <div class="task ${task.completed ? 'completed' : ''}">
+    <input type="checkbox" ${task.completed ? 'checked' : ''}id="${task.index}">
+    <label for = "${task.index}"> "${task.description}"</label>
+    <div class="remove"><button class="btnremove"><i class="fa">&#xf014;</i></button></div>
+    <div class="remove"><button class="btnEdit"><i class="fa-solid fa-pen-to-square"></i></button></div>
+
+    </div> <hr>
+    `).join('');
+  const removeTask = (index) => {
+    for (let i = index + 1; i < tasks.length; i += 1) {
+      tasks[i].index -= 1;
+    }
+    tasks.splice(index, 1);
+    taskPopulate(tasks);
+    setStorage(tasks);
+  };
+
+  const btnRemove = document.querySelectorAll('.btnremove');
+  if (btnRemove) {
+    btnRemove.forEach((btn, index) => {
+      btn.addEventListener('click', () => {
+        removeTask(index);
+      });
+    });
+  }
+  const editTask = (index) => {
+    const label = document.querySelector(`label[for='${index}']`);
+    const currentText = label.innerText.trim();
+    const input = document.createElement('input');
+    input.type = 'text';
+    input.className = 'txtInput';
+    input.value = currentText;
+    label.replaceWith(input);
+    input.focus();
+    const saveChanges = () => {
+      const newText = input.value.trim();
+      if (newText !== '' && newText !== currentText) {
+        const newLabel = document.createElement('label');
+        newLabel.htmlFor = index;
+        newLabel.innerText = newText;
+        input.replaceWith(newLabel);
+        tasks[index].description = newText;
+        setStorage();
+      } else {
+        label.replaceWith(label);
+      }
+    };
+    input.addEventListener('blur', saveChanges);
+    input.addEventListener('keydown', (event) => {
+      if (event.key === 'Enter') {
+        event.preventDefault();
+        saveChanges();
+      } else if (event.key === 'Escape') {
+        event.preventDefault();
+        label.replaceWith(label);
+      }
+    });
+  };
+
+  const btnEdit = document.querySelectorAll('.btnEdit');
+  if (btnEdit) {
+    btnEdit.forEach((btn, index) => {
+      btn.addEventListener('click', () => {
+        editTask(index);
+      });
+    });
+  }
+};
+
+export const addList = () => {
+  const taskDescription = taskEntry.value;
+
+  const newTask = {
+
+    description: taskDescription,
+    completed: false,
+    index: tasks.length,
+  };
+
+  tasks.push(newTask);
+  setStorage(tasks);
+  taskPopulate(tasks);
+  taskEntry.value = '';
+};
```

Comment:
- [ ] Elements were added to the array and to local storage successfully, but we missed one of the project requirements for task arrays. The property index should be set to the value of the new array length, meaning that if I have ten elements, the last element index should be 10 and not 9.

![array](https://user-images.githubusercontent.com/53124548/236433045-273fe126-c85b-43c2-bced-0bdbb68f4771.png)

![index](https://user-images.githubusercontent.com/53124548/236433099-c630518e-f0e0-4ccf-b0f7-67a53f5f7258.png)